feat(client): pass arguments to the flow language server

kumarharsh opened the pull request:
- Let users specify if they want the LSP to try loading
  the flow-bin from the local node_modules folder

Closes #3

Commit:
feat(client): pass arguments to the flow language server

- Let users specify if they want the LSP to try loading
  the flow-bin from the local node_modules folder

Closes #3

jbachhardie commented:
I'm concerned that this configuration method is a bit too low-level and error-prone (users might easily mis-spell arguments). Considering there's a fairly limited list of options that the language server accepts:

```js
{
  flowPath: argv['flow-path'],
  tryFlowBin: argv['try-flow-bin'],
  autoDownloadFlow: !argv['no-auto-download'],
};
```

I'd rather we had configuration options for `flowPath`, `tryFlowBin` and `autoDownloadFlow` that could benefit then from autocompletion and typechecking in the config file.

kumarharsh commented:
I did it this way because then the plugin won't have to update everytime flow-language-server adds some config on their end. That being said, I do see your point. I'll make the changes soon.

kumarharsh commented:
Actually, maybe the third option will have to be `noAutoDownload` because we're basically passing the arguments directly to the CLI, and this way we won't be doing a double inversion.

Commit:
fix(client): make the options mirror the cli ones

Based on this discussion: https://github.com/jbachhardie/vscode-flow-ls/pull/4#issuecomment-394191754

jbachhardie commented:
Thanks for making those changes.

This looks good, I'll merge and publish.
